### PR TITLE
Support em-http-request

### PR DESCRIPTION
**Pull request by minsikzzang**
Support em-http-request

Test
* ruby 1.8.7-p375 - partially passed (cannot perform full tests because of lack of fiber support)
* ruby 1.9.3-p484 - partially passed (cannot perform full tests because of issue with thread / fiber)
* ruby 2.0.0-p481 - fully passed
* ruby 2.1.1 - fully passed
* ruby 2.1.2 - fully passed

### CODE REVIEW

File: lib/new_relic/agent/http_clients/em_http_request_wrappers.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        def type
+          "EMHTTPRequest"
+        end
+
+        def host
+          self["host"] || self["Host"] || @uri.host
+        end
+
+        def method
+          (@request.method || "GET").to_s.upcase
```
Comment (jamesjoshuahill) [inline lib/new_relic/agent/http_clients/em_http_request_wrappers.rb]: Minor point: to me it feels odd to call `.to_s.upcase` on `"GET"`.

File: lib/new_relic/agent/http_clients/em_http_request_wrappers.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        def [](key)
+          @request.headers[key]
+        end
+
+        def []=(key, value)
+          @request.add_header(key, value)
+        end
+
+        def uri
```
Comment (jamesjoshuahill) [inline lib/new_relic/agent/http_clients/em_http_request_wrappers.rb]: Could you use `attr_reader`?

**Comment by jamesjoshuahill** [comment]
:dolphin: for effort! Not sure my comments are much help, sorry.

**Comment by minsikzzang** [comment]
Very helpful :+1:

**Pull request closed without merging**